include/ccl_haloprofile.h

```diff
@@ -0,0 +1,54 @@
+/** @file */
+#ifdef __cplusplus
+extern "C" {
```

Comment:
It would be nice if you can switch to the new include guards and use CCL_BEGIN_DECLS / CCL_END_DECLS following the example in 

https://github.com/LSSTDESC/CCL/blob/master/include/ccl_correlation.h